notebooks/exploratory-analysis/UMAP-aggregated-fs-profiles/nbconverted/umap-analysis.r

```diff
@@ -0,0 +1,179 @@
+    shape = "Plate"
+  ) +
+  guides(fill = guide_legend(override.aes = list(color = control_colors))) +  # Add color to legend
+  theme(
+    legend.position = "right",  # Place legend on the right
+    legend.text = element_text(size = 16),  # Increase legend text size
+    legend.title = element_text(size = 18, face = "bold"),  # Bold legend title
+    axis.text = element_text(size = 16),  # Increase axis text size
+    axis.title = element_text(size = 18, face = "bold"),  # Bold axis labels
+    plot.title = element_text(size = 20, face = "bold", hjust = 0.5)  # Centered title
+  )
+
+
+# save figure into the output folder
+path.figure <- file.path(path.output, "umap_control_profiles.png")
+ggsave(path.figure, width = 15, height = 11, units = "in", dpi = 300)
+print(paste("UMAP of control profiles saved at", path.figure))
+
+# Extract only morphology data (exclude metadata)
+morphology_features <- morphology_df
+
+# Ensure row names in metadata match morphology_features
+rownames(metadata_df) <- rownames(morphology_features)
+
+# Perform UMAP
+umap_result <- umap(morphology_features, n_components = 2)
+
+# Create a data frame with UMAP results and metadata
+umap_df <- data.frame(
+  UMAP1 = umap_result$layout[,1],
+  UMAP2 = umap_result$layout[,2],
+  Plate = metadata_df$Metadata_plate_barcode,
+  Pathway = metadata_df$Metadata_Pathway
+)
+
+# set seed
+set.seed(0)
+
+# Generate a color palette dynamically for unique plates
+plate_colors <- setNames(
+  colorRampPalette(brewer.pal(8, "Dark2"))(length(unique(umap_df$Plate))),
+  unique(umap_df$Plate)
+)
+
+# Plot UMAP with enhanced aesthetics for publication
+ggplot(umap_df, aes(x = UMAP1, y = UMAP2, color = Plate)) +
+  geom_point(alpha = 0.7, size = 5, shape = 16) +
+  scale_color_manual(values = plate_colors) +
+  theme_minimal(base_size = 18) +
+  labs(
+    title = "UMAP of profiles across all plates",
+    x = "UMAP 1",
+    y = "UMAP 2",
+    color = "Plate"
+  ) +
+  theme(
+    legend.position = "right",
+    legend.text = element_text(size = 18),
+    legend.title = element_text(size = 20, face = "bold"),
+    axis.text = element_text(size = 18),
+    axis.title = element_text(size = 20, face = "bold"),
+    plot.title = element_text(size = 22, face = "bold", hjust = 0.5)
+  )
+
+  # save figure into the output folder
+path.figure <- file.path(path.output, "umap_all_plate_profiles.png")
+ggsave(path.figure, width = 15, height = 11, units = "in", dpi = 300)
+print(paste("UMAP of all plate profiles saved at", path.figure))
+
+
+# Generate a color palette dynamically for unique pathways (colorblind-friendly)
+pathway_colors <- setNames(
+  colorRampPalette(brewer.pal(8, "Dark2"))(length(unique(umap_df$Pathway))),
+  unique(umap_df$Pathway)
+)
+
+# Plotting UMAP coloring based on the Pathway
+ggplot(umap_df, aes(x = UMAP1, y = UMAP2, color = Pathway)) +
+  geom_point(alpha = 0.7, size = 5, shape = 16) +
+  scale_color_manual(values = pathway_colors) +
+  theme_minimal(base_size = 18) +
```

Comment:
IMO `theme_bw()` is better (less distracting than `theme_minimal()` and easier to see with white background)